src/textual/widget.py

```diff
@@ -3283,15 +3294,42 @@ def remove(self) -> AwaitRemove:
         await_remove = self.app._remove_nodes([self], self.parent)
         return await_remove
 
-    def remove_children(self) -> AwaitRemove:
-        """Remove all children of this Widget from the DOM.
+    def remove_children(
```

Comment:
This method looks like it applies to children and their descendants? Basically equivalent to `self.query(selector).remove()`, so there doesn't seem much point in this existing.

What we could probably use is a method to remove only direct children, which would better serve the use case of removing and mounting in a batch.

Reply:
Understood. See 808f536.